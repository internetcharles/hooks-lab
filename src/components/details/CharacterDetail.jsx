import React from 'react';
import PropTypes from 'prop-types';

const CharacterDetail = ({ name, image, description }) => (
  <figure data-testid='detail'>
    <h1>{name}</h1>
    <img src={image} alt={name} />
    <h3>{description}</h3>
  </figure>
);

CharacterDetail.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default CharacterDetail;