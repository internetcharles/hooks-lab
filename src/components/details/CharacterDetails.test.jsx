import React from 'react';
import { render } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';

describe('CharacterDetail component', () => {
  it('renders character', () => {
    const { asFragment } = render(<CharacterDetail 
    name="John"
    image="john.jpg"
    description="John is great."
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});