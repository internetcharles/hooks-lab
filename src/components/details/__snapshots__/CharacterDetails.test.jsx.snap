// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders character 1`] = `
<DocumentFragment>
  <figure
    data-testid="detail"
  >
    <h1>
      John
    </h1>
    <img
      alt="John"
      src="john.jpg"
    />
    <h3>
      John is great.
    </h3>
  </figure>
</DocumentFragment>
`;
